tests/cpp/core/LoggingUtilityTest.cpp

```diff
@@ -0,0 +1,200 @@
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#define BOOST_TEST_MODULE LoggingUtilityTest
+
+#include <boost/make_shared.hpp>
+#include <boost/regex.hpp>
+#include <boost/test/unit_test.hpp>
+namespace ut = boost::unit_test;
+
+#include "DisplayGroup.h"
+#include "DummyContent.h"
+#include "LoggingUtility.h"
+#include "rest/RestLogger.h"
+
+#include <QObject>
+
+namespace
+{
+const QSize wallSize( 1000, 1000 );
+const std::string regexJson{
+R"(\{
+    "event": \{
+        "count": 2,
+        "last_event": "contentWindowAdded",
+        "last_event_date": "\d{4}-\d{2}-\d{2}\u\d{2}:\d{2}:\d{2}.\d{6}"
+    \},
+    "window": \{
+        "accumulated_count": 2,
+        "count": 2,
+        "date_set": "\d{4}-\d{2}-\d{2}\u\d{2}:\d{2}:\d{2}.\d{6}"
+    \}
+\}
+)"
+};
+const std::string defaultJson{
+R"({
+    "event": {
+        "count": 0,
+        "last_event": "",
+        "last_event_date": ""
+    },
+    "window": {
+        "accumulated_count": 0,
+        "count": 0,
+        "date_set": ""
+    }
+}
+)"
+};
+}
+
+BOOST_AUTO_TEST_CASE( testAccumulatedWindowCount )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+    
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    ContentWindowPtr window2 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+
+    BOOST_CHECK( logger.get()->getAccumulatedWindowCount() == 0);
+    displayGroup->addContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getAccumulatedWindowCount() == 1);
+    
+    displayGroup->addContentWindow( window2 );
+    BOOST_CHECK( logger.get()->getAccumulatedWindowCount() == 2);
+
+}
+
+BOOST_AUTO_TEST_CASE( testWindowCount )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowRemoved,
+                      logger.get(), &LoggingUtility::contentWindowRemoved );
+    BOOST_CHECK( logger.get()->getWindowCount() == 0);
+    displayGroup->addContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount() == 1);
+    displayGroup->removeContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount() == 0);
+    displayGroup->removeContentWindow( window1 );
```

Comment:
this does not correctly test that the counter cannot be negative, because removing the same window a second time does not emit a windowRemoved signal. Do a separate boost test case where you add a window before connecting the windowAdded signal, then remove the window.